tests/specs/infinite-gas.k

```diff
@@ -44,19 +50,32 @@ module INFINITE-GAS-JAVA [kast]
     rule I1          <Int I2 +Int I3 => I1 -Int I3  <Int I2         requires #isConcrete(I1) andBool #isConcrete(I3) [simplification]
     rule I1         <=Int I2 +Int I3 => I1 -Int I3 <=Int I2         requires #isConcrete(I1) andBool #isConcrete(I3) [simplification]
 
-    rule ((I1 +Int I2) +Int I3) +Int I4 => (I1 +Int I3) +Int (I2 +Int I4) requires #isConcrete(I2) andBool #isConcrete(I4) andBool notBool (#isConcrete(I1) orBool #isConcrete(I3)) [simplification]
-    rule ((I1 +Int I2) -Int I3) +Int I4 => (I1 -Int I3) +Int (I2 +Int I4) requires #isConcrete(I2) andBool #isConcrete(I4) andBool notBool (#isConcrete(I1) orBool #isConcrete(I3)) [simplification]
-    rule (I1 +Int I2) -Int (I3 +Int I4) => (I1 -Int I3) +Int (I2 -Int I4) requires #isConcrete(I2) andBool #isConcrete(I4) andBool notBool (#isConcrete(I1) orBool #isConcrete(I3)) [simplification]
+    rule (I1 +Int I2) +Int I3 => (I1 +Int I3) +Int I2 requires #isConcrete(I2) andBool notBool #isConcrete(I3) [simplification]
+    rule (I1 +Int I2) -Int I3 => (I1 -Int I3) +Int I2 requires #isConcrete(I2) andBool notBool #isConcrete(I3) [simplification]
+    rule (I1 -Int I2) +Int I3 => (I1 +Int I3) -Int I2 requires #isConcrete(I2) andBool notBool #isConcrete(I3) [simplification]
+    rule (I1 -Int I2) -Int I3 => (I1 -Int I3) -Int I2 requires #isConcrete(I2) andBool notBool #isConcrete(I3) [simplification]
+
+    rule (I1 +Int I2) +Int I3 => I1 +Int (I2 +Int I3) requires #isConcrete(I2) andBool #isConcrete(I3) andBool notBool #isConcrete(I1) [simplification]
+    rule (I1 +Int I2) -Int I3 => I1 +Int (I2 -Int I3) requires #isConcrete(I2) andBool #isConcrete(I3) andBool notBool #isConcrete(I1) [simplification]
+    rule (I1 -Int I2) +Int I3 => I1 +Int (I3 -Int I2) requires #isConcrete(I2) andBool #isConcrete(I3) andBool notBool #isConcrete(I1) [simplification]
+    rule (I1 -Int I2) -Int I3 => I1 -Int (I2 +Int I3) requires #isConcrete(I2) andBool #isConcrete(I3) andBool notBool #isConcrete(I1) [simplification]
 endmodule
 
 module INFINITE-GAS-COMMON
     imports EVM
 
  // Generic rules
 
-    rule I -Int I => 0                                              [simplification]
-    rule I1 -Int (I1 -Int I2) => I2                                 [simplification]
-    rule (notBool (A andBool B)) andBool A => (notBool B) andBool A [simplification]
+    rule (notBool (A andBool B)) andBool A => (notBool B) andBool A                      [simplification]
+    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification]
+
+    rule I -Int I => 0              [simplification]
+    rule (I1 -Int I2) +Int I2 => I1 [simplification]
+
+    rule I1 +Int (I2 +Int I3) => (I1 +Int I2) +Int I3 [simplification]
+    rule I1 +Int (I2 -Int I3) => (I1 +Int I2) -Int I3 [simplification]
+    rule I1 -Int (I2 +Int I3) => (I1 -Int I2) -Int I3 [simplification]
+    rule I1 -Int (I2 -Int I3) => (I1 -Int I2) +Int I3 [simplification]
```

Comment:
this combined with lines 34-37 may lead to infinite loop

Reply:
Otherwise all good.

Reply:
I've adjusted the rules to make it so that infinite loops cannot happen I think, please have another look @denis-bogdanas